src/NuGet.Core/NuGet.Packaging/Signing/Utility/CertificateChainUtility.cs

```diff
@@ -15,19 +15,20 @@ public static class CertificateChainUtility
         /// <summary>
         /// Create a list of certificates in chain order with the leaf first and root last.
         /// </summary>
-        /// <remarks>This must not be used on timestamp certificates as this method does not enforce the requirement
-        /// that timestamps be trusted.</remarks>
         /// <param name="certificate">The certificate for which a chain should be built.</param>
         /// <param name="extraStore">A certificate store containing additional certificates necessary
         /// for chain building.</param>
         /// <param name="logger">A logger.</param>
+        /// <param name="certificateType">The certificate type.</param>
         /// <exception cref="ArgumentNullException">Thrown if <paramref name="certificate" /> is <c>null</c>.</exception>
         /// <exception cref="ArgumentNullException">Thrown if <paramref name="extraStore" /> is <c>null</c>.</exception>
         /// <exception cref="ArgumentNullException">Thrown if <paramref name="logger" /> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="certificateType" /> is undefined.</exception>
         public static IReadOnlyList<X509Certificate2> GetCertificateChainForSigning(
```

Comment:
This method is no longer only for signing but also for timestamping. Should we update the name?